site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+# GEP-709: Cross Namespace References from Routes
+
+* Issue URL: https://github.com/kubernetes-sigs/gateway-api/issues/709
+* Status: Implementable
+
+## TLDR
+
+This GEP attempts to tackle both cross namespace forwarding and route inclusion.
+These are closely related concepts that can be solved with a new ReferencePolicy
+resource that enables app admins to describe where they trust references from.
+
+## Motivation/User Journeys/Background
+
+This GEP enables the following capabilities:
+
+1. Retaining full control of Gateway and Routes in an infra namespace, while
+   targeting apps in different namespaces.
+1. Traffic splitting between Services in different namespaces.
+1. Mesh overrides to target Services in different namespaces. 
+
+## ReferencePolicy
+
+Anytime we allow crossing a namespace boundary, we need to be very cautious. To
+ensure that this feature is safe, we need to enforce a handshake mechanism that
+requires resources in both namespaces to agree to this reference. To accomplish
+that, a new ReferencePolicy resource should be introduced.
+
+![Reference Policy](images/709-referencepolicy.png)
+
+With this model, Routes would be able to directly reference Routes and Services
+in other namespaces. These references would only be considered valid if a
+ReferencePolicy in the target namespace explicitly allowed it. 
+
+The following example shows how a HTTPRoute in namespace foo could reference
+a Service in namespace bar. In this example a ReferencePolicy in the bar
+namespace explicitly allows references to Services from HTTPRoutes in the foo
+namespace.
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  rules:
+  - matches:
+    - path: /bar
+    forwardTo:
+      backend:
+      - name: bar
+        namespace: bar
+---
+kind: ReferencePolicy
+metadata:
+  name: bar
+  namespace: bar
+spec:
+  from:
+  - group: networking.gateway.k8s.io
+    kind: HTTPRoute
+    namespace: foo
+  to:
+  - group: core
+    kind: Service
+```
+
+### API
+This proposed API is fairly straightforward, but comes with a couple notable
+decisions:
+
+1. Each ReferencePolicy only supports a single From and To section. Additional
+   trust relationships can be modeled with additional ReferencePolicy resources.
+1. Resource names are intentionally excluded from this policy for simplicity and
+   because they rarely provide any meaningful protection.
```

Comment:
... as the owner of the namespace can always rename resources/change the content of the resources to match a given policy.

Reply:
Good call, added this extra context.